pkg/nsx/services/nsxserviceaccount/cluster.go

```diff
@@ -250,7 +269,15 @@ func (s *NSXServiceAccountService) createPIAndCCP(normalizedClusterName string,
 		}
 		s.PrincipalIdentityStore.Add(&pi)
 	} else if !hasPI != (piObj == nil) {
+		log.Error(fmt.Errorf("old PI exists %v %v", !hasPI, piObj == nil), "")
```

Comment:
Better to add the variable name in the log message, too. After some time, no one knows "old PI exists true false" means, especially the first values is negation of hasPI, it will easily confuses people. I'd like to suggest `"old PI exists: '!hasPI' = 'true/false', 'piObj==nil' = 'true/false'"`.

By the way, I think it's better to leverage the existing structured log formatter of log.Error. Like:
```
log.Error(fmt.Errorf("conflicting old PI exists"), "Failed to create PI and CCPN", "!hasPI", !hasPI, "piObj == nil", piObj == nil)
```

Reply:
done